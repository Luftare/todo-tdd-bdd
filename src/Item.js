import React from 'react';

export default ({ name, deleteItem }) => (
  <li>
    {name}
    <button onClick={deleteItem} data-delete-item-unit-test>
      x
    </button>
  </li>
);
