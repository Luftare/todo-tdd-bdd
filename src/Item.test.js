import React from 'react';
import { mount } from 'enzyme';

import Item from './Item';

describe('Item', () => {
  let component;
  let deleteItemMock;

  beforeEach(() => {
    deleteItemMock = () => {};
    component = mount(<Item name="some text" deleteItem={deleteItemMock} />);
  });

  it('renders correctly', () => {
    expect(component).toMatchHtmlSnapshot();
  });

  it('should display item name', () => {
    expect(component).toIncludeText('some text');
  });

  it('button for deleting item within, should know how to delete item', () => {
    const deleteButton = component.find('button[data-delete-item-unit-test]');

    expect(deleteButton).toHaveProp('onClick', deleteItemMock);
  });
});
