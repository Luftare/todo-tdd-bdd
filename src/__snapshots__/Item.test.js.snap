// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders correctly 1`] = `
<li>
  some text
  <button
    data-delete-item-unit-test="true"
  >
    x
  </button>
</li>
`;
